### PR TITLE
Added the pybindings for Extend_Connectivity_Data

### PR DESCRIPTION
**Pull request by aasthabagree21**
## Proposed changes
Added the pybindings for Extend_Connectivity_Data function and generated a small test to ensure that the python fucntion is called correctly.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/IO/H5/Python/VolumeData.cpp [hunk 1]
```diff
@@ -49,7 +50,16 @@ void bind_h5vol(py::module& m) {
            py::arg("observation_id"))
       .def("get_data_by_element", &h5::VolumeData::get_data_by_element,
            py::arg("start_observation_value"), py::arg("end_observation_value"),
-           py::arg("components_to_retrieve") = std::nullopt);
+           py::arg("components_to_retrieve") = std::nullopt)
+      .def("extend_connectivity_data_1D",
```
Comment (nilsvu) [inline src/IO/H5/Python/VolumeData.cpp]: use lower case: `_1d`, `_2d`, `_3d`

File: src/IO/H5/Python/VolumeData.cpp [hunk 2]
```diff
@@ -6,6 +6,7 @@
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 #include <string>
+#include <iostream>
```
Comment (nilsvu) [inline src/IO/H5/Python/VolumeData.cpp]: delete

File: src/IO/H5/Python/ExtendConnectivityData.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Distributed under the MIT License.
```
Comment (nilsvu) [inline src/IO/H5/Python/ExtendConnectivityData.py]: Please add only the bindings in this PR, not the CLI

**Review by nilsvu — COMMENTED** [review]
You can squash the fixups

File: src/IO/H5/Python/ExtendConnectivityData.py (hunk 3 above)
Comment (aasthabagree21) [reply src/IO/H5/Python/ExtendConnectivityData.py]: Is there any simple way that I can delete this file from the PR?

Comment (knelli2) [reply src/IO/H5/Python/ExtendConnectivityData.py]: You'll have to do an interactive rebase and then "edit" the first commit. We can do a hangout if that's helpful

Comment (aasthabagree21) [reply src/IO/H5/Python/ExtendConnectivityData.py]: I think a hangout would be helpful. Thank you!

File: tests/Unit/IO/H5/Test_VolumeData.py [hunk 4]
```diff
@@ -276,13 +284,17 @@ def test_offset_and_length_for_grid(self):
...
+                grid_name="[B0(L1I0,L1I0,L1I0)]",
                 all_grid_names=all_grid_names,
                 all_extents=all_extents,
             ),
             (0, 8),
         )
 
+    def test_extend_connectivity_data_3D(self):
+        obs_ids = self.vol_file.list_observation_ids()
+        self.vol_file.extend_connectivity_data_3d(obs_ids)
```
Comment (nilsvu) [inline tests/Unit/IO/H5/Test_VolumeData.py:299]: Can you add a small check that this did something? Like check the size of the connectivity dataset or something like that.

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Review by nilsvu — APPROVED** [review]

**Review by knelli2 — APPROVED** [review]

**Pull request merged**